client/client_test.go

```diff
@@ -0,0 +1,138 @@
+package client_test
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/heptio/sonobuoy/client"
+)
+
+const (
+	zeroEightResults = "test_data/results-0.8.tar.gz"
+	zeroNineResults  = "test_data/results-0.9.tar.gz"
+	zeroTenResults   = "test_data/results-0.10.tar.gz"
+)
+
+var allResultVersionFiles = []string{
+	zeroEightResults,
+	zeroNineResults,
+	zeroTenResults,
+}
+
+func TestNamespacedResources(t *testing.T) {
+	testCases := []struct {
+		testName         string
+		namespace        string
+		expectedServices []string
+	}{
+		{
+			testName:         "no services in the default namespace",
+			namespace:        "default",
+			expectedServices: []string{},
+		},
+		{
+			testName:  "some services in the kube-system namespace",
+			namespace: "kube-system",
+			// a non exhaustive list
+			expectedServices: []string{"kube-dns", "kube-proxy", "kube-apiserver"},
+		},
+	}
+
+	// Run the test for every version
+	for _, resultsFilename := range allResultVersionFiles {
+		t.Run(fmt.Sprintf("results filename: %v", resultsFilename), func(t *testing.T) {
+			results, err := client.NewResults(resultsFilename)
+			if err != nil {
+				t.Fatal("")
+			}
+			for _, tc := range testCases {
+				defaultServices := [][]byte{}
+				err = results.NamespacedResources(func(path string, info os.FileInfo, err error) error {
+					//TODO(chuckha): not super happy the consumer has to have a path here to filter.
+					// The other option is parsing enough of the data to figure out if they care about it.
+					if path == tc.namespace+"/Services.json" {
+						reader, ok := info.Sys().(io.Reader)
+						if !ok {
+							return errors.New("info.Sys() is not an io.Reader")
+						}
+						fileData, err := ioutil.ReadAll(reader)
+						if err != nil {
+							return err
+						}
+						fmt.Println("services.json:", fileData)
+						defaultServices = append(defaultServices, fileData)
+					}
+					return err
+				})
+				if len(defaultServices) != len(tc.expectedServices) {
+					t.Fatalf("expected %v services found %v", len(tc.expectedServices), len(defaultServices))
+				}
+			}
+		})
+	}
+}
+
+// func TestHosts(t *testing.T) {
```

Comment:
if this test compiles, but doesn't pass, you can use `t.Skip` to return early from the test and avoid checking in commented code.

Reply:
ah yeah, it doesn't compile, I only finished converting the first function and have to come back to these.